torch/distributed/rpc/api.py

```diff
@@ -449,19 +446,23 @@ def _invoke_rpc(to, func, rpc_type, args=None, kwargs=None):
             rpc_type,
             str(qualified_name) if qualified_name is not None else func.__qualname__,
             get_worker_info().name,
-            info.name,
+            dst_worker_info.name,
         )
 
     args = args if args else ()
     kwargs = kwargs if kwargs else {}
 
     if qualified_name is not None:
-        fut = _invoke_rpc_builtin(info, qualified_name, rf, *args, **kwargs)
+        fut = _invoke_rpc_builtin(dst_worker_info, qualified_name, rf, *args, **kwargs)
+    elif isinstance(func, torch.jit.ScriptFunction):
+        fut = _invoke_rpc_torchscript(
+            dst_worker_info.name, torch._jit_internal._qualified_name(func), *args, **kwargs
```

Comment:
are you planning to pass "rf" (autograd profiler) as well in this diff or follow up diff?

Reply:
Yes. With this streamlining, it's easier to extend profiler to the new code path.